api.planx.uk/modules/file/service/uploadFile.ts

```diff
@@ -59,20 +61,21 @@ const buildFileUrl = (key: string, path: "public" | "private") => {
 export function generateFileParams(
   file: Express.Multer.File,
   filename: string,
+  filekey?: string,
 ): {
   params: S3.PutObjectRequest;
   fileType: string | null;
   key: string;
 } {
   const fileType = getType(filename);
-  const key = `${nanoid()}/${filename}`;
+  const key = `${filekey || nanoid()}/${filename}`;
 
   const params = {
     ACL: process.env.AWS_S3_ACL,
     Key: key,
-    Body: file.buffer,
+    Body: file?.buffer || JSON.stringify(file),
     ContentDisposition: `inline;filename="${filename}"`,
-    ContentType: file.mimetype,
+    ContentType: file?.mimetype || "application/json",
```

Comment:
A bit surprisingly, `exportData` (our `DigitalPlanningApplication` typed object) satisfies type `Express.Multer.File` expected by these functions. It won't have a `buffer` & `mimetype` properties, but the `S3.PutObjectRequest.Body` accepts any string so I've just defined this simple `JSON.stringify()` fallback here.

Would be a good one to more intentionally revisit in a follow-up PR if this grows beyond a proof-of-concept!